### DISCUSSION TITLE
how can i pass query parmas in router.push?

### DISCUSSION THREAD

**Discussion by parth25** [47ng/nuqs#505] [General] [answered]
```
const flightFilterSearchBoxSubmit = () => {
  router.push('/flight/flightDetails').then(() => {
  
  }
}
```

**Comment by franky47** [comment, marked as answer]
You can use a `URL` object and set the search params on it, or you can use the serializer helper to pass it your state values as JS entities (numbers, dates, object etc) and it will do the right serialisation for you while keeping it type-safe:

https://nuqs.47ng.com/docs/utilities#serializer-helper
